docs/edgeql/select.rst

```diff
@@ -1026,6 +1026,70 @@ exclusively fetch the ``Movie.characters`` of type ``Hero``.
     ...
   }
 
+Accessing types in polymorphic queries
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While the type of an object is displayed inside polymorphic queries, this
+is simply a convenience of the REPL and not a property that can be accessed.
+This is particularly noticeable if you cast an object to ``json``, making it
+impossible to determine the type if the query is polymorphic:
+
+.. code-block:: edgeql-repl
+
+    db> select Person limit 1;
+    {default::Villain {id: 6c22bdf0-5c03-11ee-99ff-dfaea4d947ce}}
+    db> select <json>Person limit 1;
+    {Json("{\"id\": \"6c22bdf0-5c03-11ee-99ff-dfaea4d947ce\"}")}
+
+The type of an object is found inside ``__type__`` which is a link that
+carries various information about a type, including its ``name``.
+
+.. code-block:: edgeql-repl
+
+    db> select <json>Person { 
+    ...  __type__: {
+    ...    name 
+    ...    } 
+    ...  } limit 1;
+    {Json("{\"__type__\": {\"name\": \"default::Villain\"}}")}
+
+This information can be pulled into the top level by assigning a name to
+the ``name`` property inside ``__type__``:
+
+    db> select <json>Person { n := .__type__.name } limit 1;
+    {Json("{\"n\": \"default::Villain\"}")}
+
+There is nothing magical about ``__type__``; it is a simple link to an object
+called ``ObjectType`` that leads to all of the possible information to know
+about any type in the database. The splat operator can be used to see this
+object's makeup, while the double splat operator produces too much input to
+show on this page. Playing around with the splat and double splat operator
+inside ``__type__`` is a quick way to get some insight into the internals
+of EdgeDB.
```

Comment:
Shifted this slightly to center more around this object's `__type__` link rather than about the `ObjectType` type since we're starting this description from `__type__`. Information is basically the same but the perspective has been shifted.
```suggestion
There is nothing magical about ``__type__``; it is a simple link to an object
of the type ``ObjectType`` which contains all of the possible information to
know about the type of the current object. The splat operator can be used to
see this object's makeup, while the double splat operator produces too much
output to show on this page. Playing around with the splat and double splat
operator inside ``__type__`` is a quick way to get some insight into the
internals of EdgeDB.
```